### PR TITLE
Build for hip gpu backends

### PR DESCRIPTION
**Pull request by williamberman**
Issue: https://github.com/flexflow/FlexFlow/issues/345

#### Testing the build
```sh
FF_GPU_BACKEND=hip_rocm ./docker/build.sh
```

#### Current status
hip_rocm builds e2e with a few changes to legion and flexflow source

#### Small source modifications for build
misc small changes to the source to get the build working that should be ok to merge.

#### Move tools to top level directory
We glob for files under src to get the source files for the flexflow
target. Moving tools to the top level directory prevents the tools
sourcefiles from accidentally being added to the flexflow target
source files.

change substitution_to_dot cuda_add_executable to add_executable.
When building with hip_rocm, we don't have cuda available and shouldn't
need to build with it for substitution_to_dot as the target does

#### Remaining:
- [x] hip_rocm backend builds
- [x] cuda backend builds
- [x] Do changes to legion source have to be merged in
- [x] Is switching from miopen.h to miopen/miopen.h header acceptabe
- [x] Fix out of date hip kernels to match kernel headers
- [x] Feedback on hip kernel changes
- [x] update Dockerfile to conditionally install hip dependencies
- [x] Document Docker build changes

#### Misc

Additional note on the legion change, I also don't know if the const_cast to remove the volatile qualifier is _sound_ in that context. I mainly added it to get legion compiling with the changed build config

### CODE REVIEW

File: CMakeLists.txt
Comment (williamberman) [inline CMakeLists.txt:245]: @lockshaw This is a bit hacky. Alternatively we commit a script into source that does the same thing. Do you have a preference?

Comment (lockshaw) [reply CMakeLists.txt:245]: I'd probably lean toward the cmake approach, but I don't have a particularly strong feeling either way. Is there any difference in terms of impact on developer experience between the two?

Comment (williamberman) [reply CMakeLists.txt:245]: I don't think there's a difference in developer experience, both are grockable. +1 to cmake approach, will leave as is

File: src/ops/concat.cpp
Comment (williamberman) [inline src/ops/concat.cpp:46]: @lockshaw I think these kernels are out of date. They don't match the header

File: src/ops/fused.cpp
Comment (williamberman) [inline src/ops/fused.cpp:169]: @lockshaw I copy pasted from fused.cu. I'm not familiar with what's being asserted and if it should still be from fused.cpp

File: .gitmodules
Comment (lockshaw) [inline .gitmodules:3]: Why is this necessary? I'd rather not point master at your fork of legion

File: CMakeLists.txt
Comment (lockshaw) [inline CMakeLists.txt:250]: Is there any documentation on how to get these libraries installed and setup? Would be nice to automate this as much as possible

File: config/config.inc
Comment (lockshaw) [inline config/config.inc:119]:
```suggestion
if [[ -n "$ROCM_PATH" ]]; then
```

File: docker/build.sh
Comment (lockshaw) [inline docker/build.sh:36]: Any risk of this having spaces?

File: include/flexflow/graph.h
Comment (lockshaw) [inline include/flexflow/graph.h:87]: What's the reason for this change?

File: python/flexflow_c.cc
Comment (lockshaw) [inline python/flexflow_c.cc:1737]: Why extra parens?

File: src/ops/fused.cpp
Comment (lockshaw) [reply src/ops/fused.cpp:169]: Yeah this is correct. This is a change from unity: we currently model replication with an additional replica dimension, so every tensor dimension internally ends up one bigger

**Review by lockshaw — APPROVED** [review]
In general looks good! I have a few questions it would nice to get answered, but I didn't see anything blocking so once CI passes I'm down to merge as it's unlikely to break more than it currently is :slightly_smiling_face:.

Might be good to have @eddy16112 also do a pass at some point, but I don't think we need to hold up merging for that.

EDIT: actually I think we need to resolve why we need a separate legion branch first, the others shouldn't be blocking but that one is.

File: src/runtime/accessor.cc
Comment (lockshaw) [inline src/runtime/accessor.cc:319]: Why is this necessary?

File: src/parallel_ops/partition.cpp
Comment (lockshaw) [inline src/parallel_ops/partition.cpp:60]: Why are these necessary?

File: .gitmodules
Comment (williamberman) [reply .gitmodules:3]: Yeah don't plan to merge this yet. Still need feedback on if the source change to legion is necessary

**Comment by lockshaw** [comment]
@eddy16112 @jiazhihao (adding Wei and Zhihao for discussing legion changes)

File: CMakeLists.txt
Comment (williamberman) [reply CMakeLists.txt:250]: The dockerfile has the install steps/necessary pacakges! The steps there should be replicatable on a standard ubuntu instance. Should we link to them from source and/or the readme?

File: include/flexflow/graph.h
Comment (williamberman) [reply include/flexflow/graph.h:87]: There is a static assert in legion that breaks if we don't cap the buffer

File: python/flexflow_c.cc
Comment (williamberman) [reply python/flexflow_c.cc:1737]: Without the parens, the compiler interprets it as a function prototype

File: src/runtime/accessor.cc
Comment (williamberman) [reply src/runtime/accessor.cc:319]: Standard explicit template instantiation to fix missing symbols during linking. Not sure _why_ the symbols are missing but erred on side of just adding to get working -- worst case scenario binary is sligtly bigger?

File: src/parallel_ops/partition.cpp
Comment (williamberman) [reply src/parallel_ops/partition.cpp:60]: Same as https://github.com/flexflow/FlexFlow/pull/392#discussion_r1013560722

**Comment by williamberman** [comment]
> In general looks good! I have a few questions it would nice to get answered, but I didn't see anything blocking so once CI passes I'm down to merge as it's unlikely to break more than it currently is 🙂.
> 
> Might be good to have @eddy16112 also do a pass at some point, but I don't think we need to hold up merging for that.

Let's wait until we at least figure out if the legion changes are necessary before merging :grin:

File: docker/build.sh
Comment (williamberman) [reply docker/build.sh:36]: good catch, will quote

File: config/config.inc
Comment (williamberman) [reply config/config.inc:119]: Other conditionals in script use single bracket conditionals -- should we just keep it consistent?

**Comment by williamberman** [comment]
@gabrieleoliaro can you take a look at the docker changes? Want to make sure that's in sync with your CI work :)

File: docker/base/Dockerfile
Comment (williamberman) [inline docker/base/Dockerfile:48]: @gabrieleoliaro Thought for this change is that if I'm building a container I should be building from my local repo. Does that work with you?

File: config/config.inc
Comment (lockshaw) [reply config/config.inc:119]: Yeah match the style of the script

File: include/flexflow/graph.h
Comment (lockshaw) [reply include/flexflow/graph.h:87]: I guess the question more was why was this change not necessary before?

File: src/runtime/accessor.cc
Comment (lockshaw) [reply src/runtime/accessor.cc:319]: I guess my concern here is that I think (@jiazhihao correct me if I'm wrong) we don't actually have support for double-precision tensors (everything is `DT_FLOAT`) so I'm concerned that if something is requiring this that the code requiring it may be incorrect

Comment (williamberman) [reply src/runtime/accessor.cc:319]: Ah gotcha, yeah if it's that unexpected, can do some digging to try to figure out what's causing it

File: include/flexflow/graph.h
Comment (williamberman) [reply include/flexflow/graph.h:87]: Ah gotcha yeah don't know. 

https://github.com/flexflow/FlexFlow/commit/3249a2c5b4ba65b37dc621bd3a3906e2bde0c509

@jiazhihao can you provide some color on why it was commented out in this commit?

File: CMakeLists.txt
Comment (williamberman) [reply CMakeLists.txt:250]: @lockshaw added docs here https://github.com/flexflow/FlexFlow/blob/5530127939d5be7f0361ac6c849cfbd9b37b9a85/INSTALL.md

File: src/runtime/accessor.cc
Comment (jiazhihao) [reply src/runtime/accessor.cc:319]: > I guess my concern here is that I think (@jiazhihao correct me if I'm wrong) we don't actually have support for double-precision tensors (everything is DT_FLOAT) so I'm concerned that if something is requiring this that the code requiring it may be incorrect

We plan to add support for `DT_DOUBLE`, `DT_HALF`, and other data types. The current implementation supports some type casts for MT5, and our plan is to add a more systematic support for mix-precision training.

File: include/flexflow/graph.h
Comment (jiazhihao) [reply include/flexflow/graph.h:87]: This change relates to supporting serialization for very large models. Actually, I don't like the changes I made --- it's kind of a hack. `LEGION_MAX_RETURN_SIZE` specifies the maximum size of a `Future` returned by a Legion task. We want to specify it in `cmake` and the specified value should be passed to both Legion and FlexFlow compilation.

Comment (williamberman) [reply include/flexflow/graph.h:87]: @jiazhihao Maybe worth making a followup issue for that?

File: src/runtime/accessor.cc
Comment (williamberman) [reply src/runtime/accessor.cc:319]: @lockshaw sounds like the explicit template instantiation is ok then in this case?

File: .gitmodules
Comment (lockshaw) [reply .gitmodules:3]: @jiazhihao These legion changes are the main thing I want you to look at in this PR. @williamberman can provide more explanation about what the changes actually are

Comment (williamberman) [reply .gitmodules:3]: https://github.com/StanfordLegion/legion/commit/21269d0de541649efccbfe038d6515728e673af9 This is the only remaining legion change. `dom.dim` is volatile when targeting cuda or hip. I put in a const_cast to remove the volatile so it can be passed to deserialize

Comment (jiazhihao) [reply .gitmodules:3]: This legion change looks good to me.

File: include/flexflow/graph.h
Comment (jiazhihao) [reply include/flexflow/graph.h:87]: #417 tracks this issue.

Comment (williamberman) [reply include/flexflow/graph.h:87]: perf!

File: .gitmodules
Comment (williamberman) [reply .gitmodules:3]: Ok, put up here https://gitlab.com/StanfordLegion/legion/-/merge_requests/597

I believe this means we'll have to point at legion master instead of legion stable temporarily? Not sure how this works with submodules.

If so, should we make a follow up issue to point back to stable once a new stable release is cut?

File: src/runtime/accessor.cc
Comment (lockshaw) [reply src/runtime/accessor.cc:319]: It's a small enough thing that I don't want to hold up this PR on it anyway :slightly_smiling_face:

File: .gitmodules
Comment (jiazhihao) [reply .gitmodules:3]: We use a customized branch of Legion: https://gitlab.com/StanfordLegion/legion/-/tree/flexflow, which is based on Legion's control replication branch. One issue with the master branch of Legion is that it does not support control replication, a key optimization to enable scalability.

Comment (jiazhihao) [reply .gitmodules:3]: If the Legion team doesn't want the PR to be merged to the main stream, we can merge it to the `flexflow` branch.

Comment (lockshaw) [reply .gitmodules:3]: @jiazhihao Would it be possible to merge it to the flexflow branch now so we don't have to wait on the legion team? (I imagine you'd have to do that merge as I doubt @williamberman has merge permissions in legion :slightly_smiling_face:)

Comment (jiazhihao) [reply .gitmodules:3]: I have merged Will's changes to the flexflow branch of Legion.

Comment (jiazhihao) [reply .gitmodules:3]: I will create a PR to update FlexFlow's Legion submodule.

Comment (lockshaw) [reply .gitmodules:3]: Yeah @jiazhihao you probably need to use different branches for different PRs :slightly_smiling_face:

Comment (williamberman) [reply .gitmodules:3]: Awesome, thank you!

File: docker/build.sh
Comment (williamberman) [reply docker/build.sh:5]: Great catch on both, adding now!

File: tools/protobuf_to_json/CMakeLists.txt
Comment (williamberman) [reply tools/protobuf_to_json/CMakeLists.txt:17]: Sure thing!

File: tools/substitutions_to_dot/CMakeLists.txt
Comment (williamberman) [reply tools/substitutions_to_dot/CMakeLists.txt:12]: yep!

File: .gitmodules
Comment (eddy16112) [reply .gitmodules:3]: > I have merged Will's changes to the flexflow branch of Legion.
It seems to be a bug for the Legion HIP backend. Let's push the fix to the master branch?

**Comment by williamberman** [comment]
Commit pointing to legion fork has been dropped. Waiting until https://github.com/flexflow/FlexFlow/pull/425 lands, then will rebase and we're g2g

File: include/flexflow/config.h
Comment (williamberman) [inline include/flexflow/config.h:26]: @eddy16112 @jiazhihao Sorry, last question. Is changing this header path acceptable? When I install the latest rocm version, this is the path to the miopen header. Maybe `<miopen.h>` is the previous location of the header and is this change ok?

**Comment by lockshaw** [comment]
> One more small thing -- it would be really good to run shellcheck before merging this, since we're modifying bash files. We should probably add a CI workflow for this later too

Good idea. #432

**Comment by williamberman** [comment]
The CI is currently broken because the python setup.py script reads the config.linux script as a string and attempts to parse the values set in it, so it can't be treated as a regular shell script. We added some standard bash conventions so the config script so it could have default values and read overrides in from the environment which breaks the adhoc variable parsing.

File: include/flexflow/config.h
Comment (eddy16112) [reply include/flexflow/config.h:26]: If the directory of MIOPEN is changed, I am OK with your changes.

File: python/flexflow_c.cc
Comment (eddy16112) [inline python/flexflow_c.cc:1737]: why do we need another `()`?

**Comment by eddy16112** [comment]
Regarding the CI, if we can not find a AMD GPU for the CI, we can test the code with `FF_GPU_BACKEND=cuda`

File: python/flexflow_c.cc
Comment (williamberman) [reply python/flexflow_c.cc:1737]: Without the additional (), it gets parsed as a function prototype

File: include/flexflow/config.h
Comment (williamberman) [reply include/flexflow/config.h:26]: Ok great! Thank you :)

File: python/flexflow_c.cc
Comment (lockshaw) [reply python/flexflow_c.cc:1737]: https://github.com/flexflow/FlexFlow/pull/392#discussion_r1013553783

**Comment by GYDmedwin** [comment]
Hi @williamberman . I am trying to build the project according to the method you gave, but the error message said that kfd could not access the project due to insufficient permission. It may be because of this problem that the compiling error of legion was reported. I guess it is because docker permission is not enough to access kfd during the compilation process. Can you give me some advice? Thank you.
The error message is as follows:
`make[2]: *** [deps/legion/runtime/CMakeFiles/LegionRuntime.dir/build.make:237: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/legion/legion_analysis.cc.o] Error 1`
`make[2]: *** [deps/legion/runtime/CMakeFiles/LegionRuntime.dir/build.make:475: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/legion/runtime.cc.o] Error 1`
`make[1]: *** [CMakeFiles/Makefile2:390: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/all] Error 2`

**Comment by williamberman** [comment]
> Hi @williamberman . I am trying to build the project according to the method you gave, but the error message said that kfd could not access the project due to insufficient permission. It may be because of this problem that the compiling error of legion was reported. I guess it is because docker permission is not enough to access kfd during the compilation process. Can you give me some advice? Thank you. The error message is as follows: `make[2]: *** [deps/legion/runtime/CMakeFiles/LegionRuntime.dir/build.make:237: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/legion/legion_analysis.cc.o] Error 1` `make[2]: *** [deps/legion/runtime/CMakeFiles/LegionRuntime.dir/build.make:475: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/legion/runtime.cc.o] Error 1` `make[1]: *** [CMakeFiles/Makefile2:390: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/all] Error 2`

Hey @GYDmedwin I might need some additional information here :) I take it this is happening during the docker container build? What's the full error and where does it occur in the build process (maybe some more console output would be helpful). It sounds like you're running on a machine with an actual amd gpu if I'm reading your message correctly. Fwiw, we're mainly merging this with the intention of targeting amd but did not run it on an actual machine with an amd gpu due to the particulars involved in finding one that supports rocm.

**Comment by GYDmedwin** [comment]
@williamberman Thank you for your reply. Yes, I test this on a machine with the actual amd gpu MI100 and the errors occur during the docker container build. 
All the errors occur in the following step:
`Step 18/19 : RUN mkdir -p build && cd build &&     ../config/config.linux &&     make -j $N_BUILD_CORES &&     ../config/config.linux &&     make install`

And the main error's details are as follows:
`-- Using the single-header code from /usr/FlexFlow/build/_deps/json-src/single_include/
-- FlexFlow MAX_DIM: 5
Traceback (most recent call last):
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 257, in <module>
    main()
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 241, in main
    target_list = readFromKFD()
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 200, in readFromKFD
    line = f.readline()
PermissionError: [Errno 1] Operation not permitted
-- hip::amdhip64 is SHARED_LIBRARY
Traceback (most recent call last):
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 257, in <module>
    main()
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 241, in main
    target_list = readFromKFD()
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 200, in readFromKFD
    line = f.readline()
PermissionError: [Errno 1] Operation not permitted
-- hip::amdhip64 is SHARED_LIBRARY
Traceback (most recent call last):
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 257, in <module>
    main()
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 241, in main
    target_list = readFromKFD()
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 200, in readFromKFD
    line = f.readline()
PermissionError: [Errno 1] Operation not permitted
-- hip::amdhip64 is SHARED_LIBRARY
-- pybind11 v2.11.0 dev1
-- Found PythonInterp: /opt/conda/bin/python (found suitable version "3.9.13", minimum required is "3.6") 
-- Found PythonLibs: /opt/conda/lib/libpython3.9.so`

`[ 10%] Building CXX object deps/legion/runtime/CMakeFiles/RealmRuntime.dir/realm/numa/numasysif.cc.o
Traceback (most recent call last):
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 257, in <module>
    main()
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 241, in main
    target_list = readFromKFD()
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 200, in readFromKFD
    line = f.readline()
PermissionError: [Errno 1] Operation not permitted


[ 38%] Building CXX object deps/legion/runtime/CMakeFiles/RealmRuntime.dir/realm/deppart/image_4_5.cc.o
Traceback (most recent call last):
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 257, in <module>
    main()
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 241, in main
    target_list = readFromKFD()
  File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 200, in readFromKFD
    line = f.readline()
PermissionError: [Errno 1] Operation not permitted`

This error occurred in several places, and the content was the same. I only showed three. I guess because of this error, legion compilation failed, and finally FlexFlow compilation failed.
Another error message details are as follows:
`26 warnings and 1 error generated when compiling for gfx803.
make[2]: *** [deps/legion/runtime/CMakeFiles/LegionRuntime.dir/build.make:475: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/legion/runtime.cc.o] Error 1
26 warnings and 1 error generated when compiling for gfx803.
make[2]: *** [deps/legion/runtime/CMakeFiles/LegionRuntime.dir/build.make:237: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/legion/legion_analysis.cc.o] Error 1
26 warnings and 1 error generated when compiling for gfx803.
make[2]: *** [deps/legion/runtime/CMakeFiles/LegionRuntime.dir/build.make:489: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/legion/region_tree_1.cc.o] Error 1
26 warnings generated when compiling for host.
26 warnings generated when compiling for host.
26 warnings generated when compiling for host.
26 warnings generated when compiling for host.
26 warnings generated when compiling for host.
26 warnings generated when compiling for host.
make[1]: *** [CMakeFiles/Makefile2:390: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/all] Error 2
make: *** [Makefile:136: all] Error 2`

**Comment by williamberman** [comment]
> @williamberman Thank you for your reply. Yes, I test this on a machine with the actual amd gpu MI100 and the errors occur during the docker container build. All the errors occur in the following step: `Step 18/19 : RUN mkdir -p build && cd build && ../config/config.linux && make -j $N_BUILD_CORES && ../config/config.linux && make install`
> 
> **And the main error's details are as follows:** -- Using the single-header code from /usr/FlexFlow/build/_deps/json-src/single_include/ -- FlexFlow MAX_DIM: 5 Traceback (most recent call last): File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 257, in main() File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 241, in main target_list = readFromKFD() File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 200, in readFromKFD line = f.readline() PermissionError: [Errno 1] Operation not permitted -- hip::amdhip64 is SHARED_LIBRARY Traceback (most recent call last): File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 257, in main() File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 241, in main target_list = readFromKFD() File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 200, in readFromKFD line = f.readline() PermissionError: [Errno 1] Operation not permitted -- hip::amdhip64 is SHARED_LIBRARY Traceback (most recent call last): File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 257, in main() File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 241, in main target_list = readFromKFD() File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 200, in readFromKFD line = f.readline() PermissionError: [Errno 1] Operation not permitted -- hip::amdhip64 is SHARED_LIBRARY -- pybind11 v2.11.0 dev1 -- Found PythonInterp: /opt/conda/bin/python (found suitable version "3.9.13", minimum required is "3.6") -- Found PythonLibs: /opt/conda/lib/libpython3.9.so
> 
> `[ 10%] Building CXX object deps/legion/runtime/CMakeFiles/RealmRuntime.dir/realm/numa/numasysif.cc.o Traceback (most recent call last): File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 257, in main() File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 241, in main target_list = readFromKFD() File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 200, in readFromKFD line = f.readline() PermissionError: [Errno 1] Operation not permitted
> 
> [ 38%] Building CXX object deps/legion/runtime/CMakeFiles/RealmRuntime.dir/realm/deppart/image_4_5.cc.o Traceback (most recent call last): File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 257, in main() File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 241, in main target_list = readFromKFD() File "/opt/rocm-5.2.0/bin/rocm_agent_enumerator", line 200, in readFromKFD line = f.readline() PermissionError: [Errno 1] Operation not permitted`
> 
> This error occurred in several places, and the content was the same. I only showed three. I guess because of this error, legion compilation failed, and finally FlexFlow compilation failed.
> 
> **Another error message details are as follows:** 26 warnings and 1 error generated when compiling for gfx803. make[2]: *** [deps/legion/runtime/CMakeFiles/LegionRuntime.dir/build.make:475: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/legion/runtime.cc.o] Error 1 26 warnings and 1 error generated when compiling for gfx803. make[2]: *** [deps/legion/runtime/CMakeFiles/LegionRuntime.dir/build.make:237: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/legion/legion_analysis.cc.o] Error 1 26 warnings and 1 error generated when compiling for gfx803. make[2]: *** [deps/legion/runtime/CMakeFiles/LegionRuntime.dir/build.make:489: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/legion/region_tree_1.cc.o] Error 1 26 warnings generated when compiling for host. 26 warnings generated when compiling for host. 26 warnings generated when compiling for host. 26 warnings generated when compiling for host. 26 warnings generated when compiling for host. 26 warnings generated when compiling for host. make[1]: *** [CMakeFiles/Makefile2:390: deps/legion/runtime/CMakeFiles/LegionRuntime.dir/all] Error 2 make: *** [Makefile:136: all] Error 2

Ok great thank you for the extra details! I opened up an issue here https://github.com/flexflow/FlexFlow/issues/457. I think your best immediate bet would be building with the Makefiles and on your standard system (not in docker)

**Pull request merged**
